### ISSUE TITLE
Segfault at runtime when calling a function from an `@export`'d function pointer 

### ISSUE THREAD

**Issue by Jarred-Sumner** [ziglang/zig#20472]
### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

To reproduce:
1. Run `zig test` on this file:
```zig
fn function() callconv(.C) void {}
const exported = &function;
comptime {
    @export(exported, .{ .name = "exported2" });
}

extern fn exported2() void;

test {
    exported2();
}
```
2. Output:
```zig
Bus error at address 0x1029a4150
Panicked during a panic. Aborting.
error: the following test command crashed:
/Users/jarred/Code/bun/.zig-cache/o/3c26355fcac2b8e03bcdb703800c6976/test
```

This reproduces on macOS arm64 and Windows x64. It caused the "fuzzy-wuzzy.test.ts" test to fail in this [github action](https://github.com/oven-sh/bun/actions/runs/9753279129/job/26918802500) run (and locally in a build with optimizations enabled). 

- The calling convention specified doesn't seem to matter, it happens for both `.SysV` and `.C` on Windows x64.
- The `"name"` doesn't seem to matter, it happens when the name is the same as the function name

### Expected Behavior

The test should pass without crashing

**Comment by llogick** [comment]
The technical aspect of this is that the code is exporting the equivalent of
`*const *const fn () callconv(.C) void`
while the
`extern fn exported2() void;`
is expecting a
`*const fn () callconv(.C) void`.

To handle that
```zig
const exported2 = @extern(*const *const fn () callconv(.C) void, .{ .name = "exported2" });
...
	exported2.*();
...
```

**Comment by Vexu** [comment]
This footgun should be addressed by this accepted proposal #14911
